### ISSUE TITLE
[CoE Starter Kit - BUG] App Details not working

### ISSUE THREAD

**Issue by FQ365Microsoft** [microsoft/coe-starter-kit#8186]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

App Details and app permissions are not working. It was working but apparently something is broken over the weekend.
<img width="733" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/116768543/02f9bea3-5a00-4905-be34-a4e05e028a69">

### Expected Behavior

It should work and launch the app details page.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.24

### What app or flow are you having the issue with?

Power Platform Admin View

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

Just lunch the Admin View App and click on App Details

### Anything else?

Have tried 

AB#3091

**Comment by Jenefer-Monroe** [comment]
This is still working for me, and I've never seen that error before. 
- Does this repro for all records or just a few? 
- Does the app still exist? Can you manually browse to it instead? 
- Perhaps something got messed up in the record, can you please set it to run through inventory and then see if it still repros the next day after inventory runs?
![Image](https://github.com/microsoft/coe-starter-kit/assets/89584709/e8394ca6-314c-4bce-8730-27ad5e695be5)

**Comment by FQ365Microsoft** [comment]
Thanks Jenefer.
I tried debugging and found the formula is not able to expand Self.Selected.Item.'App Environment'.Name. I checked the data is present and is showing all the correct IDs and if I change the formula to Self.Selected.Item.'App Environment ID' , it works perfectly fine.

Failing formula
Launch(theAppMakerURL & "environments/" & Self.Selected.Item.'App Environment'.Name & "/apps/" & Self.Selected.Item.App & "/details"))))
Successful formula
Launch(theAppMakerURL & "environments/" & Self.Selected.Item.'App Environment ID' & "/apps/" & Self.Selected.Item.App & "/details"))))

**Comment by Jenefer-Monroe** [comment]
Thank you for debugging with me! 
I think that formula wont work for the default environment unfortunately. We use name instead of id because the urls require the default- for the envt 
This does work for me so I think there is something wrong with your data. 
Can you please view the Environment list and see if you have blanks in the name column? 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/2ddd19b1-4151-4940-a3c8-926bfba1be4a)

**Comment by FQ365Microsoft** [comment]
Hi Jenefer,
Yes I checked again and there is no issue with data and no blank Name column.
You are right Environment ID does not work for default environment. And interestingly, I just found it has started working again with the original 'App Environment'.Name. So I rolled back. Dont know what happened and it started working again. I am thinking of maybe sync was in progress but it should not have impacted the existing records Name field. Strange issue. I will keep an eye if it appears again. We can close the issue.
